File: AlleDrogo.Internal.Contracts/Command/AddAuctionCommand.cs
using AlleDrogo.Internal.Contracts.Models;
using MediatR;
using System;

namespace AlleDrogo.Internal.Contracts.Command
{
    public class AddAuctionCommand : IRequest<AddAuctionResult>
    {
        public AuctionItemModel Item { get; set; }

        public string Username { get; set; }

        public string Title { get; set; }

        public DateTime EndDate { get; set; }

        public string Description { get; set; }

        public bool IsBuyNow { get; set; }

        public decimal? BuyNowValue { get; set; }

        public decimal InitialValue { get; set; }
    }
}

File: AlleDrogo.Web/ClientApp/src/app/models/add-auction-command.ts
import { AuctionItemModel } from "./auction-item-model";

export class AddAuctionCommand {
  constructor() {
    this.item = new AuctionItemModel();
  }

  item: AuctionItemModel;
  userName: string;
  title: string;
  endDate: Date;
  description: string;
  isBuyNow: boolean;
  buyNowValue: number = null;
  initialValue: number;
}
